### PR TITLE
Reload module when we use --update option

### PR DESCRIPTION
**Pull request by saurabsa**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -1589,11 +1589,16 @@ def _install_dev_spaces_extension(extension_name):
     return True
 
 
-def _update_dev_spaces_extension(extension_name):
+def _update_dev_spaces_extension(extension_name, extension_module):
     from azure.cli.core.extension import ExtensionNotInstalledException
     try:
+        # unloading the imported module to update
+        del sys.modules[extension_module]
```
Comment (williexu) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: This is not good as it does not update any references to the old module. Also, a re-import will force python to to an expensive full import.

The way to do this more safely (though still not usually advised) is to use [reload()](https://docs.python.org/3/library/importlib.html#importlib), which will update the module currently in `sys.modules` and allow existing references to have access to the updated module. It is also more performant.

Let's discuss this offline and see how we can address your scenario in the extension.

**Comment by troydai** [comment]
Kindly remind that the cut of date is this Wednesday.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 2]
```diff
@@ -1589,11 +1589,17 @@ def _install_dev_spaces_extension(extension_name):
...
 
-def _update_dev_spaces_extension(extension_name):
+def _update_dev_spaces_extension(extension_name, extension_module):
     from azure.cli.core.extension import ExtensionNotInstalledException
     try:
         from azure.cli.command_modules.extension import custom
         custom.update_extension(extension_name=extension_name)
+
+        # reloading the imported module to update
+        from importlib import reload as reload
```
Comment (troydai) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: Hum? Can't it just be `from importlib import reload`?

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 3]
```diff
@@ -1592,13 +1592,14 @@ def _install_dev_spaces_extension(extension_name):
...
     try:
-        # unloading the imported module to update
-        del sys.modules[extension_module]
-
         from azure.cli.command_modules.extension import custom
         custom.update_extension(extension_name=extension_name)
+
+        # reloading the imported module to update
+        from importlib import reload as reload
+        reload(sys.modules[extension_module])
```
Comment (williexu) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: If you want this to also work for python2 you can do something like:

```
try:
    from importlib import reload as reload
except ImportError:
    pass
reload(module)
```

Comment (saurabsa) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: Oh, I thought it worked for Python2 as well as it passed CI. Updating ...

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py (hunk 2 above)
Comment (saurabsa) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: Updated

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py (hunk 3 above)
Comment (williexu) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: `reload()` is a builtin function for python2, hence the try-catch around the import.

**Pull request merged**